torch/csrc/autograd/python_engine.h

```diff
@@ -18,6 +19,7 @@ struct PythonEngine : public Engine {
       bool keep_graph,
       bool create_graph,
       const edge_list& outputs = {}) override;
+  virtual std::unique_ptr<AnomalyMetadata> make_anomaly_metadata() override;
```

Comment:
Why do we create the metadata objects in the engine? It doesn't seem to be the best place to put it

Reply:
Because each engine is responsible to create the metadata used to tag the objects it works with.